lib/blockchain.go

```diff
@@ -482,6 +482,13 @@ type Blockchain struct {
 	bestHeaderChain    []*BlockNode
 	bestHeaderChainMap map[BlockHash]*BlockNode
 
+	// Track the highest QC we've seen.
+	highestQC *QuorumCertificate
+
+	// Tracks all uncommitted blocks in memory
+	uncommittedBlocks    []*MsgDeSoBlock
+	uncommittedBlocksMap map[BlockHash]*MsgDeSoBlock
```

Comment:
What's interesting about this map is that you could use it to store not just the uncommitted blocks within the best chain, but also uncommitted blocks from possible re-orgs as well. 

It would be the equivalent of the `safeBlocks` map in the reference implementation. So we could pass in these blocks into the `ProcessTipBlock(tip, safeBlocks)` function of the event loop so it can properly construct timeout QCs that result in reorgs. 

Just a thought